src/analyzing/CFG_build.mli

```diff
@@ -1 +1,13 @@
 val cfg_of_stmts : IL.stmt list -> IL.cfg
+
+(* Given the language and a program ast, compute the control flow
+ * graphs of all the functions in that program.
+ *
+ * The program itself is also considered a top-level function and
+ * included in the result.
+ *
+ * For each function, returns a pair of the its parameters and its
+ * control flow graph.
+ *)
+val cfgs_in_program :
```

Comment:
I was thinking more of a visitor rather than returning a list here, but if the perf or mem usage isn't affected (we'll measure it) then I don't mind this way. However, I think the "program itself as a top-level function" should be returned separately, I don't think you want to run the `Implicit_return` pass on that no?

Maybe:

```ocaml
type fun_cfg = { 
  fparams : IL.name list;
  fcfg : IL.cfg;
 }

val cfgs_in_program : 
    Lang.t -> AST_generic.program -> fun_cfg list * fun_cfg
```
?

Reply:
Oh, interesting, I didn't think about a visitor. I'll return a list here (with your suggested changes to separate out the top-level function) but if benchmark results suggest otherwise, I can switch to the visitor.